src/shallow_copy.rs

```diff
@@ -78,6 +81,20 @@ impl<T> ShallowCopy for Vec<T> {
     }
 }
 
+#[cfg(feature = "bytes")]
+impl ShallowCopy for bytes::Bytes {
+    unsafe fn shallow_copy(&mut self) -> Self {
+        self.clone()
```

Comment:
I'm new to the `Bytes` crate, but from what I understand, since `Bytes` itself is reference-counted type, I believe a simple `clone()` here satisfies the requirements of an aliasing copy.

An alternative is `Bytes::from(self.as_ref())` but I believe this is doing the same thing.

Reply:
Ah, no, that's now quite right. With `ShallowCopy`, only one of the two shallow copies are dropped, the other is simply forgotten. That would cause the given `Bytes` to leak, since the reference count would never hit 0. Take a look at the implementation for `Arc` further up in this file -- it unsafely creates another reference without increasing the count. This is why `ShallowCopy` is so wildly unsafe!

Reply:
Updated to use `Bytes::from_static`. I'm wondering if a simple `Bytes::from(self.as_ref())` would work, but I *think* `as_ref` would probably increase the reference count, and that new reference would get passed along to the new `Bytes` instance? So using `from_static` is my attempt to completely disconnect the two `Bytes` instances.

Reply:
Hmm... I'm not familiar enough with how `Bytes` works internally to say for sure. I think in theory `from_static` should do the right thing though.

Reply:
I finally had some time today to sit down and read through the implementation of `Bytes`, and I believe `from_static` indeed satisfies the requirements for `ShallowCopy`!